beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -489,6 +532,130 @@ impl<T: EthSpec> ValidatorMonitor<T> {
+            // where the proposer who missed the block is the proposer of the block at block_root(slot)
+            if let (Ok(block_root), Ok(prev_block_root)) =
+                (state.get_block_root(slot), state.get_block_root(prev_slot))
+            {
+                // Found missed block
+                if block_root == prev_block_root {
+                    let slot_epoch = slot.epoch(T::slots_per_epoch());
+                    let prev_slot_epoch = prev_slot.epoch(T::slots_per_epoch());
+
+                    if let Ok(shuffling_decision_block) = state
+                        .proposer_shuffling_decision_root_at_epoch(
+                            slot_epoch,
+                            shuffling_decision_block,
+                        )
+                    {
+                        // Only update the cache if it needs to be initialised or because
+                        // slot is at epoch + 1
+                        if proposers_per_epoch.is_none() || slot_epoch != prev_slot_epoch {
+                            proposers_per_epoch = self.get_proposers_by_epoch_from_cache(
+                                slot_epoch,
+                                shuffling_decision_block,
+                            );
+                        }
+
+                        // Only add missed blocks for the proposer if it's in the list of monitored validators
+                        let slot_in_epoch = slot % T::slots_per_epoch();
+                        if let Some(proposer_index) = proposers_per_epoch
+                            .as_deref()
+                            .and_then(|proposers| proposers.get(slot_in_epoch.as_usize()))
+                        {
+                            let i = *proposer_index as u64;
+                            if let Some(pub_key) = self.indices.get(&i) {
+                                if self.validators.get(pub_key).is_some() {
+                                    // Add to missed blocks
+                                    self.missed_blocks.insert((slot_epoch, i, slot));
+                                    // In order to overcome the case where a validator misses a block and the cache gets pruned which might prove
+                                    // challenging for the user, we also want to keep track of all the missed blocks in total_missed_blocks
+                                    self.total_missed_blocks.insert(
+                                        i,
+                                        self.total_missed_blocks.get(&i).unwrap_or(&0) + 1,
+                                    );
+                                }
+                            }
+                        } else {
+                            debug!(
+                                self.log,
+                                "Could not get proposers for epoch {slot_epoch:?} from cache"
+                            );
```

Comment:
```suggestion
                            debug!(
                                self.log,
                                "Could not get proposers for from cache";
                                "epoch" => ?slot_epoch
                            );
```

This is more consistent with our logging elsewhere.